tests/models/test_rank_torch.py

```diff
@@ -0,0 +1,123 @@
+#  Copyright 2024 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import typing as tp
+from itertools import product
+
+import implicit.cpu
+import numpy as np
+import pytest
+import torch
+from scipy import sparse
+
+from rectools.models.rank import Distance, ImplicitRanker
+from rectools.models.rank_torch import TorchRanker
+
+T = tp.TypeVar("T")
+EPS_DIGITS = 5
+pytestmark = pytest.mark.filterwarnings("ignore:invalid value encountered in true_divide")
+
+
+def gen_rankers() -> tp.List[tp.Tuple[tp.Any, tp.Dict[str, tp.Any]]]:
```

Comment:
Let's refactor it a bit. Let it return `tp.List[tp.Union[ImplicitRanker, TorchRanker]]`. Will be easier to read all of the tests.